Mathlib/Analysis/CstarAlgebra/Module.lean

```diff
@@ -0,0 +1,270 @@
+    _ = ‖x‖ ^ 2 * ‖⟪y, y⟫‖ := by simp [norm_smul]
+    _ = ‖x‖ ^ 2 * ‖y‖ ^ 2 := by
+                simp only [norm_eq_sqrt_norm_inner_self, norm_nonneg, Real.sq_sqrt]
+    _ = (‖x‖ * ‖y‖) ^ 2 := by simp only [mul_pow]
+  refine (pow_le_pow_iff_left (R := ℝ) (norm_nonneg ⟪x, y⟫_A) ?_ (by norm_num)).mp this
+  exact mul_nonneg CstarModule.norm_nonneg CstarModule.norm_nonneg
+
+protected lemma norm_triangle [CompleteSpace A] (x y : E) : ‖x + y‖ ≤ ‖x‖ + ‖y‖ := by
+  have h : ‖x + y‖ ^ 2 ≤ (‖x‖ + ‖y‖) ^ 2 := by
+    calc _ ≤ ‖⟪x, x⟫ + ⟪y, x⟫‖ + ‖⟪x, y⟫‖ + ‖⟪y, y⟫‖ := by
+          simp only [norm_eq_sqrt_norm_inner_self, inner_add_right, inner_add_left, ← add_assoc,
+            norm_nonneg, Real.sq_sqrt]
+          exact norm_add₃_le _ _ _
+      _ ≤ ‖⟪x, x⟫‖ + ‖⟪y, x⟫‖ + ‖⟪x, y⟫‖ + ‖⟪y, y⟫‖ := by gcongr; exact norm_add_le _ _
+      _ ≤ ‖⟪x, x⟫‖ + ‖y‖ * ‖x‖ + ‖x‖ * ‖y‖ + ‖⟪y, y⟫‖ := by gcongr <;> exact norm_inner_le E
+      _ = ‖x‖ ^ 2 + ‖y‖ * ‖x‖ + ‖x‖ * ‖y‖ + ‖y‖ ^ 2 := by
+          simp [norm_eq_sqrt_norm_inner_self]
+      _ = (‖x‖ + ‖y‖) ^ 2 := by simp only [add_pow_two, add_left_inj]; ring
+  refine (pow_le_pow_iff_left CstarModule.norm_nonneg ?_ (by norm_num)).mp h
+  exact add_nonneg CstarModule.norm_nonneg CstarModule.norm_nonneg
+
+/-- This allows us to get `NormedAddCommGroup` and `NormedSpace` instances on `E` via
+`NormedAddCommGroup.ofCore` and `NormedSpace.ofCore`. -/
+lemma normedSpaceCore [CompleteSpace A] : NormedSpace.Core ℂ E where
+  norm_nonneg x := CstarModule.norm_nonneg
+  norm_eq_zero_iff x := norm_zero_iff x
+  norm_smul c x := by simp [norm_eq_sqrt_norm_inner_self, norm_smul, ← mul_assoc]
+  norm_triangle x y := CstarModule.norm_triangle x y
+
+lemma norm_eq_csSup [CompleteSpace A] (v : E) :
+    ‖v‖ = sSup { ‖⟪w, v⟫_A‖ | (w : E) (_ : ‖w‖ ≤ 1) } := by
+  let instNACG : NormedAddCommGroup E := NormedAddCommGroup.ofCore normedSpaceCore
+  let instNS : NormedSpace ℂ E := .ofCore normedSpaceCore
+  refine Eq.symm <| IsGreatest.csSup_eq ⟨⟨‖v‖⁻¹ • v, ?_, ?_⟩, ?_⟩
+  · simpa only [norm_smul, norm_inv, norm_norm] using inv_mul_le_one_of_le le_rfl (by positivity)
+  · simp [norm_smul, CstarModule.inner_self_eq_norm_sq, pow_two, ← mul_assoc]
+  · rintro - ⟨w, hw, rfl⟩
+    calc _ ≤ ‖w‖ * ‖v‖ := norm_inner_le E
+      _ ≤ 1 * ‖v‖ := by gcongr
+      _ = ‖v‖ := by simp
+
+end norm
+
+section NormedAddCommGroup
+
+variable {A E : Type*} [NonUnitalNormedRing A] [StarRing A] [CstarRing A] [PartialOrder A]
+  [StarOrderedRing A] [NormedSpace ℂ A] [SMul Aᵐᵒᵖ E] [CompleteSpace A]
+  [NormedAddCommGroup E] [NormedSpace ℂ E] [StarModule ℂ A] [CstarModule A E] [IsScalarTower ℂ A A]
+  [SMulCommClass ℂ A A]
+
+/-- The function `⟨x, y⟩ ↦ ⟪x, y⟫` bundled as a continuous sesquilinear map. -/
+noncomputable def innerSL : E →L⋆[ℂ] E →L[ℂ] A :=
+  LinearMap.mkContinuous₂ (innerₛₗ : E →ₗ⋆[ℂ] E →ₗ[ℂ] A) 1 <| fun x y => by
+    simp [innerₛₗ_apply, norm_inner_le E]
+
+lemma innerSL_apply {x y : E} : innerSL x y = ⟪x, y⟫_A := rfl
```

Comment:
Unless there's a reason not to:
```suggestion
@[simp]
lemma innerSL_apply {x y : E} : innerSL x y = ⟪x, y⟫_A := rfl
```

Reply:
I think I'd rather not: I've used these lemmas backwards more often than forwards, so that I could use the properties of the bundled object. I don't think it's likely to show up out of the blue since we're pretty much always using the bare function.